### PR TITLE
feat: annotation dictionary

### LINKED ISSUE #82: Save annotations in dict
In order to avoid annotating the same song several times, we should construct a dict which maps song_id to values gathered, for example 
("song_id" : [[arousal],[valence]])

This dict should be updated every time a new song is annotated.
Before asking the gui to annotate a specific song_id, this dict should be checked beforehand. If the dict already contains that song_id then the gui doesnt need to be called.

### PR DESCRIPTION
**Pull request by tediiieee**
Annotation class for saving annotations made by gui in a json file. 

Closes #82

### CODE REVIEW

File: src/phase_utils.py [hunk 1]
```diff
@@ -324,3 +318,72 @@ def print_id_name_for_learning_profiles(learning_profiles):
     print("id -- Learning Profile")
     for lp in learning_profiles:
         print("id:", lp.get_id(), "--", f"{lp.get_name(True, False)}")
+
+
+class AnnotationStation:
```
Comment (MattiasBeming) [inline src/phase_utils.py:323]: Move AnnotationStation up in the file, and place it with the other classes

File: src/phase_utils.py [hunk 2]
```diff
@@ -324,3 +318,72 @@ def print_id_name_for_learning_profiles(learning_profiles):
...
+    (with annotations for song_ids = 1, 2):
+    {'1': [[arousal1], [valence1]], '2': [[arousal2], [valence2]]}
+    """
+
+    def __init__(self, path: Path):
+        """
+        AnnotationStation constructor
+
+        Args:
+            path (pathlib.Path): Path to dictionary in json format.
```
Comment (MattiasBeming) [inline src/phase_utils.py:339]: docstring: `(pathlib.Path)`
write `(Path)` instead

File: src/phase_utils.py [hunk 3]
```diff
@@ -1,10 +1,9 @@
 from enum import Enum
+from pathlib import Path
+import json
+import numpy as np
 
 
-###############################################################################
-# Utilitiy
-###############################################################################
-
```
Comment (MattiasBeming) [inline src/phase_utils.py:7]: Personally i think the dividers are visually pleasing. 
But existing ones could be replaced by, for example:

```
###
# Classes
###
```

```
###
# Functions
###
```

Or

Split utility into 2 files. 
If this approach is taken, then make sure all code works.

**Review by MattiasBeming — CHANGES_REQUESTED** [review]

File: src/phase_utils.py [hunk 4]
```diff
@@ -205,8 +208,76 @@ def __str__(self):
...
+    def __init__(self, path: Path):
+        """
+        AnnotationStation constructor
+
+        Args:
+            path (Path): Path to dictionary in json format.
+                    (Note: include .json tag.)
+
+        Raises:
+            FileNotFoundError: Raises error when path is not correct.
```
Comment (sporreking) [inline src/phase_utils.py:231]: "Raises error" already specified. Skip this i.e. "Raises: FileNotFoundError: When the path is not valid."

**Review by sporreking — CHANGES_REQUESTED** [review]
Overall very good! all comments are about tiny details with docstrings. They should be quick to fix.

**Review by sporreking — APPROVED** [review]
Amazing!

File: src/phase_utils.py [hunk 5]
```diff
@@ -205,8 +208,80 @@ def __str__(self):
...
+        AnnotationStation constructor.
+
+        Args:
+            path (Path): Path to dictionary in json format.
+                    (Note: include `.json` tag.)
+
+        Raises:
+            FileNotFoundError: When path is not correct.
+        """
+        self.path = path
```
Comment (MattiasBeming) [inline src/phase_utils.py:233]: `self.path` should use '_' to indicate an internal class variable:
`self._path`. 
Same goes for all class variables.

**Review by MattiasBeming — CHANGES_REQUESTED** [review]

**Review by MattiasBeming — APPROVED** [review]

**Pull request merged**